provider/pkg/gen/overrides.go

```diff
@@ -188,6 +189,40 @@ var resourceNamePropertyOverrides = map[string]string{
 	"run/v1:Service.servicesId":                               "metadata.name",
 }
 
+// schemaPropertyTypeOverrides is a map of overrides for properties of a schema type.
+var schemaPropertyTypeOverrides = map[string]map[string]*schema.TypeSpec{
+	// API ref: https://cloud.google.com/service-infrastructure/docs/service-management/reference/rest/v1/services.configs#backendrule
+	"BackendRule": {
+		// The `overridesByRequestProtocol` property in `BackendRule`,
+		// while absent, in the API reference docs, is present in the
+		// discover doc that we fetch.
+		//
+		// The property is incorrectly referring to its parent type
+		// causing a recursion. This recursion is accidental as the
+		// description of the property suggests that it's a map
+		// of the protocol and the address. So we override the
+		// type spec for this property with the correct
+		// AdditionalProperties type spec.
+		//
+		// Discovery doc: https://servicemanagement.googleapis.com/$discovery/rest?version=v1
+		"overridesByRequestProtocol": {
```

Comment:
Hopefully, the above comment is self-explanatory. If you have a different approach to fixing this or think that the recursive type is actually valid in this case, please let me know and I can revert this. However, that would mean that a solution for the recursive reference would be needed for this provider's schema (perhaps similar to k8s' ObjectMeta?)

Reply:
I discovered this issue when I ran `make build`. Here's how the property is defined in the discovery doc:

```json
"BackendRule": {
...
...
    "overridesByRequestProtocol": {
        "additionalProperties": {
          "$ref": "BackendRule"
        },
        "description": "The map between request protocol and the backend address.",
        "type": "object"
     },
...
...
}
```

Reply:
> The safest thing to do would probably be to omit this property entirely until someone can take the time to explore the behavior of this api

This is done.